### PR TITLE
[stable/jenkins] update plugin versions

### PR DESCRIPTION
**Pull request by torstenwalter**
#### What this PR does / why we need it:

It just updates used default plugins and sidecar container.

#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
  - fixes #

#### Special notes for your reviewer:
I added missing changelog entry from #22593

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[stable/mychartname]`)

### CODE REVIEW

File: stable/jenkins/CHANGELOG.md [hunk 1]
```diff
@@ -5,6 +5,21 @@ numbering uses [semantic versioning](http://semver.org).
...
+## 1.25.0 Update plugin versions & sidecar container
+
+| plugin                | old version | new version |
+| --------------------- | ----------- | ----------- |
+| kubernetes            | 1.25.3      | 1.25.7      |
+| workflow-job          | 2.38        | 2.39        |
+| credentials-binding   | 1.21        | 1.22        |
+| configuration-as-code | 1.39        | 1.41        |
+
+configAutoReload container updated from `kiwigrid/k8s-sidecar:0.1.132` to `kiwigrid/k8s-sidecar:0.1.144`
```
Comment (wmcdona89) [inline stable/jenkins/CHANGELOG.md:17]: looks like you missed the `k8s-sidecar` update in `values.yaml` :-)

**Review by wmcdona89 — CHANGES_REQUESTED** [review]

File: stable/jenkins/CHANGELOG.md (hunk 1 above)
Comment (torstenwalter) [reply stable/jenkins/CHANGELOG.md:17]: thanks for catching this. Forgot to add it ;-)

Comment (wmcdona89) [reply stable/jenkins/CHANGELOG.md:17]: there's a merge conflict line in the CHANGELOG which can be removed later
`>>>>>>> master`

**Pull request merged**